dockstore-client/src/main/java/io/dockstore/client/cli/nested/WESLauncher.java

```diff
@@ -95,38 +85,46 @@ protected void addFilesToWorkflowAttachment(List<File> workflowAttachment, File
             exceptionMessage(e, "Unable to get workflow attachment files from zip file " + zippedEntry.getName(), IO_ERROR);
         }
 
-        // Put file names in workflow attachment list
-        File[] listOfFiles = tempDir.listFiles();
-        for (int i = 0; i < listOfFiles.length; i++) {
-            if (listOfFiles[i].isFile()) {
-                // TODO check file extension (type) only for a local entry; pass all files found in remote entry folder?
-                // Locate code that grabs all imports for a non local entry and use that instead of checking extension
-                // since CWL can import many file types
-                // There may be confidential or large files that are not needed in a local directory that should
-                // not be sent to a remote endpoint?
-                if (fileIsCorrectType(listOfFiles[i])) {
-                    System.out.println("Adding file " + listOfFiles[i].getName() + " to workflow attachment");
-                    File fileToAdd = new File(tempDir, listOfFiles[i].getName());
-                    workflowAttachment.add(fileToAdd);
-                } else {
-                    System.out.println("File " + listOfFiles[i].getName() + " is not the correct type for the workflow so it will not be "
-                            + "added to the workflow attachment");
-                }
-            } else if (listOfFiles[i].isDirectory()) {
-                System.out.println("Found directory " + listOfFiles[i].getName());
+        try {
+
+            String[] fileFilter = null;
+            // For a local entry restrict the kinds of files that can be added to the workflow attachment. CWL can import many
+            // file types but there may be confidential or large files in a local directory that should
+            // not be sent to a remote endpoint.
+            // TODO Locate code that grabs all imports for a non local entry and use that instead of checking extension
+            if (abstractEntryClient.isLocalEntry()) {
+                fileFilter = new String[]{"yml", "yaml", "json", this.languageType.toString().toLowerCase()};
+            }
+            Iterator it = FileUtils.iterateFiles(tempDir, fileFilter, true);
+            while (it.hasNext()) {
+                File afile = (File) it.next();
+                workflowAttachment.add(afile);
             }
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Not sure if that matches Dockstore's style

Reply:
It doesn't, but the next two lines are sufficient so this can be straight up deleted